student.rb

```diff
@@ -1,12 +1,18 @@
 require_relative 'person'
 
 class Student < Person
-  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+  attr_reader :classroom
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
     super(age, name: name, parent_permission: parent_permission)
-    @classroom = classroom
   end
 
   def play_hooky
     '¯\(ツ)/¯'
   end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    @classroom.students.push(self) unless classroom.students.include?(self)
+  end
```

Comment:
👏 Good attempt with the method! 👏 However, when running it, an error surfaces: `NoMethodError` in `student.rb` at line 16. The error suggests an undefined method `students` for `"first":String`. Let's resolve this to ensure your code functions smoothly:

- [ ] **Undefined Method Error:** The error indicates an undefined method `students` for a string. Review your code in `classroom.rb` and ensure that the `students` method is correctly defined. Confirm that it's called on an instance of the `classroom` class and not on a string.
![image](https://github.com/MostafaShahat2021/OOP-school-library/assets/55840999/eb11d8b4-a393-43de-a3b9-401fdc1dcac6)

By addressing this issue, your method should work seamlessly. Keep up the good work! 🛠️🚀

Reply:
Hey @shahadat3669, thanks for your suggestion, I'll try to fix that 😃